### PR TITLE
Add Zoom support and misc improvements

### PR DESCRIPTION
**Pull request by potatoeggy**
This PR performs the following:

 - removes redundant checks when dropping links
 - adopts a more Pythonic style in changed sections
 - supports zoom links requiring some sort of password added via regex and will immediately fire off zoom links on first visit without doing driver checks

It does not:

 - support checking if zoom links are actually open — zoom has auto-refresh so this should be less of an issue
 - support dropping zoom links except by submitting the whole link to be dropped

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -124,16 +124,15 @@ async def on_message(message):
     else: await ch.send("If you wanna learn how to get <:pingo:822111531063836712>s, use `pingo help pls`.")
 
   #drop a link from the queue
-  if len(tokens) >= 2 and tokens[0] == "drop" and re.search(r"^[a-z0-9]{9,10}$",tokens[1]):
-    link = None
+  if len(tokens) >= 2 and tokens[0] == "drop":
     for curLink in linkQueue[key]:
-      if curLink[1] == "https://meet.google.com/lookup/" + tokens[1]:  # link matches code
+      if curLink[1] == "https://meet.google.com/lookup/" + tokens[1] or curLink[1] == tokens[1]:  # link matches code
         link = curLink
```
Comment (lakshy-gupta) [inline main.py]: What is the purpose of not removing `link = curlink`?

Comment (potatoeggy) [reply main.py]: it was there before — i don't know what it does ¯\\\_(ツ)_/¯

File: main.py [hunk 2]
```diff
@@ -124,16 +124,15 @@ async def on_message(message):
...
-    if link is None:
-      await ch.send("Link not found in queue.")
+        dropLinks[key].append(link)
+        await ch.send("dropping link to " + link[2] + "'s class from the queue. (Please allow some time for this to take effect.)")
+        break
     else:
-      dropLinks[key].append(link)
-      await ch.send("dropping link to " + link[2] + "'s class from the queue. (Please allow some time for this to take affect.)")
+      await ch.send("Link not found in queue.")
```
Comment (NoYouNotUrUncle) [inline main.py:135]: why
you just made the error message worse I do not `u n d e r s t a n d`

File: main.py [hunk 3]
```diff
@@ -124,16 +124,15 @@ async def on_message(message):
     else: await ch.send("If you wanna learn how to get <:pingo:822111531063836712>s, use `pingo help pls`.")
 
   #drop a link from the queue
-  if len(tokens) >= 2 and tokens[0] == "drop" and re.search(r"^[a-z0-9]{9,10}$",tokens[1]):
-    link = None
+  if len(tokens) >= 2 and tokens[0] == "drop":
     for curLink in linkQueue[key]:
```
Comment (lakshy-gupta) [inline main.py]:
```suggestion
    for link in linkQueue[key]:
```

File: main.py (hunk 1 above)
Comment (lakshy-gupta) [inline main.py]:
```suggestion
      if link[1] == "https://meet.google.com/lookup/" + tokens[1] or link[1] == tokens[1]:  # link matches code
```

**Comment by lakshy-gupta** [comment]
I think this PR would be better structured to support any given message (as a reminder) instead of just zoom links. This would eliminate the need for all your checks that see if a link is a zoom link or a meet link, since you would just check if a link isn't a meet link, while also providing more utility. You should simply provide a warning that the given link hasn't been recognized as a meet link. 

Also, your PR doesn't seem to be doing anything to stop the bot from visiting the zoom website, which is not ideal since the bot is not going to be designed or tested to handle sites apart from meet links.

**Comment by potatoeggy** [comment]
> I think this PR would be better structured to support any given message (as a reminder) instead of just zoom links. This would eliminate the need for all your checks that see if a link is a zoom link or a meet link, since you would just check if a link isn't a meet link, while also providing more utility. You should simply provide a warning that the given link hasn't been recognized as a meet link.

This required a bigger rewrite than I was willing to do what with special Meet link checking and handling elsewhere and etc. :P

> Also, your PR doesn't seem to be doing anything to stop the bot from visiting the zoom website, which is not ideal since the bot is not going to be designed or tested to handle sites apart from meet links.

The check can certainly be moved before but I felt it was too difficult to do so without affecting the conditional flow negatively. By default, Zoom opens up 
image
which is fairly static and free from breakage.

**Comment by lakshy-gupta** [comment]
> This required a bigger rewrite than I was willing to do what with special Meet link checking and handling elsewhere and etc. :P

It is not.

**Comment by potatoeggy** [comment]
> > This required a bigger rewrite than I was willing to do what with special Meet link checking and handling elsewhere and etc. :P
> 
> It is not.

In any event, this PR is designed only to offer Zoom support, and so other sites are out of scope. If this is contrary to future development plans, feel free to close the PR.

**Pull request closed without merging**